### PR TITLE
timeout command

### LINKED ISSUE #24: /scrim-timeout
Command to assign the scrim timeout role to a user with a countdown (minutes/hours/days?)

### PR DESCRIPTION
**Pull request by PaulMndn**
timeout command implemented and tested.

closes #24

### CODE REVIEW

File: scrimbot/timeoutlist.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        self.bot = guild.bot
+        l = []
+        for u in users:
+            task = self.bot.loop.create_task(self._timeout_runout(u["user_id"]))
+            l.append(self.User(u["user_id"],
+                               datetime.fromisoformat(u["timeout"]),
+                               task))
+        super().__init__(l)
+    
+    def __contains__(self, item):
```
Comment (jaimevisser) [inline scrimbot/timeoutlist.py]: Having __contains__ work on id while the rest of the functions in this class work on the user tuple breaks the principle of least astonishment. https://en.wikipedia.org/wiki/Principle_of_least_astonishment

File: scrimbot/timeoutlist.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
+from collections import namedtuple
+import discord
+import asyncio
+from datetime import datetime, timedelta, timezone
+
+
+class TimeoutList(list):
```
Comment (jaimevisser) [inline scrimbot/timeoutlist.py]: Having the class extend list gives it a lot of methods that need not be exposed. In this case composition over inheritance would be the better choice. https://en.wikipedia.org/wiki/Composition_over_inheritance

Comment (jaimevisser) [reply scrimbot/timeoutlist.py]: To clarify, the class that manages timeouts needs a list, but does not need to _be_ a list.

File: scrimbot/timeoutlist.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    Snowflake = namedtuple("Snowflake", "id")
+
+    def __init__(self, guild, users):
+        self.guild = guild
+        self.bot = guild.bot
+        l = []
+        for u in users:
+            task = self.bot.loop.create_task(self._timeout_runout(u["user_id"]))
+            l.append(self.User(u["user_id"],
+                               datetime.fromisoformat(u["timeout"]),
```
Comment (jaimevisser) [inline scrimbot/timeoutlist.py]: All other stored times are using timestamps, would be more consistent to use those here as well.

**Review by jaimevisser — CHANGES_REQUESTED** [review]
See inline comments.

File: scrimbot/timeoutlist.py (hunk 3 above)
Comment (PaulMndn) [reply scrimbot/timeoutlist.py]: Alright. My idea was to make it more humanly readable, but I'll change it.

File: scrimbot/timeoutlist.py (hunk 1 above)
Comment (PaulMndn) [reply scrimbot/timeoutlist.py]: removed and rewritten

File: scrimbot/timeoutlist.py (hunk 2 above)
Comment (PaulMndn) [reply scrimbot/timeoutlist.py]: implemented the principle

**Comment by PaulMndn** [comment]
Thanks for the links btw :)

**Review by jaimevisser — APPROVED** [review]

**Comment by PaulMndn** [comment]
before you merge, I'm working on another change I'd like to include.

**Pull request merged**